### DISCUSSION TITLE
QueryInterface on Texture2d giving 'not supported' error.

### DISCUSSION THREAD

**Discussion by Gavin-Williams** [amerkoleci/Vortice.Windows#424] [General]
I'm working on text rendering and trying to render to a texture. But when I request the IDXGISurface of the texture which is required for CreateDxgiSurfaceRenderTarget() I get an error

> SharpGen.Runtime.SharpGenException: 'HRESULT: [0x80004002], Module: [General], ApiCode: [E_NOINTERFACE/No such interface supported], Message: [No such interface supported]

Here's the code:
```
public static void Invoke(string text, ID3D11Texture2D target)
{
    IDXGISurface1 dxgiSurface = ID3D11Texture2D.QueryInterface<IDXGISurface1>(target);
```
The value of the texture description is:

> ArraySize = 1
> BindFlags = ShaderResource | RenderTarget
> CPUAccessFlags = None
> Format = R16G16B16A16_Float
> Size = 800 x 600
> MipLevels = 1
> MiscFlags = None
> SampleDesc = {1,0}
> Usage = Default

Is there something obviously wrong here about my target texture with respect to text rendering? Or do I need to fetch the interface in a special way? It looks correct to me.

**Comment by amerkoleci** [comment]
Please don't open issues for questions and use discussions for that

What are you trying to achieve? Rendering into Direct2D1? If so I suppose you'll need bgra8/rgba8 format, follow this link for more info: https://learn.microsoft.com/en-us/windows/win32/direct2d/direct2d-and-direct3d-interoperation-overview

**Comment by Gavin-Williams** [comment]
I've compiled a sample against your samples framework which I believe should work. The error I get on that line now is..
> System.NotSupportedException: 'Built-in COM has been disabled via a feature switch. See https://aka.ms/dotnet-illink/com for more information.'

If you can get this sample to work, please integrate it into your samples.

https://1drv.ms/u/s!AhbcRtciy6_0kJke1oPfT7qFojm78Q?e=aQjYKW

**Reply by Gavin-Williams** [reply]
I've made a pull-request if you want to accept the code that way.

**Reply by amerkoleci** [reply]
Saw that, will merge it tomorrow, thanks!

**Comment by Gavin-Williams** [comment]
There is still the issue that I can't get the interface though.

```
[DrawTextApp]
IDXGISurface1 dxgiSurface = ID3D11Texture2D.QueryInterface<IDXGISurface1>(target); **EXCEPTION**
-->
[ComObject]
public static T QueryInterface<
#if NET6_0_OR_GREATER
        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)]
#endif
    T>(object comObject) where T : ComObject =>
        As<T>(Marshal.GetIUnknownForObject(comObject));
```
Gives the error:
> System.NotSupportedException: 'Built-in COM has been disabled via a feature switch. See https://aka.ms/dotnet-illink/com for more information.'

If I add this to the project property group
> \<BuiltInComInteropSupport\>true\</BuiltInComInteropSupport\>

I get the first error I was seeing
> SharpGen.Runtime.SharpGenException: 'HRESULT: [0x80004002], Module: [General], ApiCode: [E_NOINTERFACE/No such interface supported], Message: [No such interface supported
]'

**Reply by amerkoleci** [reply]
Can you share an example with full code please?

**Reply by Gavin-Williams** [reply]
That is the sample I submitted. As far as I can tell it should work.

**Reply by amerkoleci** [reply]
Here is the working example: 
https://github.com/amerkoleci/Vortice.Windows.Samples/blob/main/src/Direct3D11/10_Text/DrawTextApp.cs#L138

I've added also this attribute to csproj:

https://github.com/amerkoleci/Vortice.Windows.Samples/blob/main/src/Direct3D11/10_Text/10_Text.csproj#L6
